### PR TITLE
Imageservice V2: implement tasks Get method

### PR DESCRIPTION
**Pull request by ozerovandrei**
Implement Get method for the Imageservice tasks.
Add unit test with documentation example.

For #1143

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

API
https://github.com/openstack/glance/blob/stable/queens/glance/api/v2/tasks.py#L121
https://github.com/openstack/glance/blob/stable/queens/glance/api/v2/tasks.py#L309

DB
https://github.com/openstack/glance/blob/stable/queens/glance/db/sqlalchemy/api.py#L1559

### CODE REVIEW

File: openstack/imageservice/v2/tasks/results.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+// GetResult represents the result of a Get operation. Call its Extract
+// method to interpret it as a Task.
+type GetResult struct {
+	commonResult
+}
+
+// TaskStatus represents valid task status.
+type TaskStatus string
```
Comment (jtopjian) [inline openstack/imageservice/v2/tasks/results.go]: Let's move this + the consts to `requests.go`

File: openstack/imageservice/v2/tasks/results.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+// Task represents a single task of the OpenStack Image service.
+type Task struct {
+	// ID is a unique identifier of the task.
+	ID string `json:"id"`
+
+	// Type represents the type of the task.
+	Type string `json:"type"`
+
+	// Status represents current status of the task.
+	Status TaskStatus `json:"status"`
```
Comment (jtopjian) [inline openstack/imageservice/v2/tasks/results.go]: This should be a `string`

**Comment by ozerovandrei** [comment]
@jtopjian 
Oh well I've gotten those custom types all wrong then. I've been thinking that they're here to allow only well-known values (`IPv4` and `IPv6` for the `IPVersion` for example).

As for documentation. I think that we need at least add inline docs for the fields that can have pre-defined well known values.
In that case those people who are checking `godoc` or reading those comments will know that they can use additional well-known values for some fields.

I've updated all PRs and they're ready.

**Comment by jtopjian** [comment]
> I've been thinking that they're here to allow only well-known values (IPv4 and IPv6 for the IPVersion for example)

Right - I used to think the same, but then I noticed that some fields are _not_ like this. Upon further investigation and discussion, those fields were accidentally glanced over and need to be cleaned up at some point. I still accidentally let one slip by every now and then. I'll try to comb through Gophercloud and get them all fixed at once. 

> As for documentation. I think that we need at least add inline docs for the fields that can have pre-defined well known values.

Good point.

> I've updated all PRs and they're ready.

Sounds good. I'll review when I have a moment. Thank you!

**Pull request merged**